lib/puppet/provider/iis_site/webadministration.rb

```diff
@@ -38,6 +38,59 @@ def create
     return exists?
   end
 
+  def update
+    Puppet.err 'UPDATE FUNCTION'
+
+    # Instantiate server manager object and get matching site using resource name
+    inst_cmd = []
+    inst_cmd << '[Void][Reflection.Assembly]::LoadWithPartialName("Microsoft.Web.Administration");'
+    inst_cmd << '$serverManager = New-Object Microsoft.Web.Administration.ServerManager;'
+    inst_cmd << "$site = $serverManager.Sites | where { $_.Name -eq '#{@resource[:name]}' };"
+    inst_cmd << "$app = $site.Applications | where { $_.Path -eq '/' };"
```

Comment:
These where clauses need to be tested on servers with more than a few sites. In prior testing with instances and create/delete, these methods get significantly slower as the number of sites go up